scripts/generate-patch

```diff
@@ -29,18 +29,20 @@ for f in packages/*; do
 			if [[ $fields -eq '0' ]]; then
 				fields='1'
 			fi
+			rm -rf ${f}/charts-original
+			mkdir -p ${f}/charts-original
 			if [[ $type == 'git' ]]; then
 				mkdir -p /tmp/tmp-charts
 				git clone --depth=1 --no-tags $url /tmp/tmp-charts
 				pwd=$(pwd)
 				cd /tmp/tmp-charts
+				git fetch --tags
 				git fetch origin $commit
 				git checkout $commit
 				cd $pwd
-				cp -r /tmp/tmp-charts/${subdirectory} ${f}/charts-original
+				cp -r /tmp/tmp-charts/${subdirectory}/* ${f}/charts-original
 				rm -rf /tmp/tmp-charts
 			else
-				mkdir -p ${f}/charts-original
 				curl -sLf ${url} | tar xvzf - -C ${f}/charts-original --strip ${fields} ${subdirectory} > /dev/null 2>&1
 			fi
```

Comment:
IMO it's worth thinking about making a commit like https://github.com/rancher/charts/pull/642/commits/9971a14071b8fe32a148f98365d8d15ee3cde10a that just pulls the `prepare-upstream` logic from `prepare` and `generate-patch` into the one script that gets called by both in order to avoid maintaining the exact same logic twice in our codebase.

Not sure if that's something we want to add into the scripts pre-2.5 (in a separate PR), but I've tested that the fix works as expected when I wrote the stuff to help me generate the patch for this PR as part of https://github.com/rancher/charts/pull/642.

cc: @cbron @StrongMonkey

Reply:
I don't know enough about generate-patch script but just looking at the scripts I agree the common part can be pulled into a single one and referenced in both places